### DISCUSSION TITLE
How to get clangd with --completion-style=detailed

### DISCUSSION THREAD

**Discussion by sbeyer** [mason-org/mason.nvim#488] [Q&A]
I am looking for a way to use `clangd` with its `--completion-style=detailed` option. I have no idea how to achieve that. Yes, I checked the `setup_handlers()` doc and the mason source, but still no idea. (Btw: What I *actually* want is to get my C++ completions work for overloads, too, and there seem to be additional packages for this, but I prefer to just get the overloads listed in a flat list, and I hope setting the completion-style option will do that.)

Any ideas?

**Comment by kbenzie** [comment]
For others who came here wanting to find this out, it can be done with the follow when using [neovim/nvim-lspconfig](https://github.com/neovim/nvim-lspconfig):

```lua
require('lspconfig').clangd.setup({
  cmd = { 'clangd', '--completion-style=detailed' }
})
```

And when using [williamboman/mason-lspconfig.nvim](https://github.com/williamboman/mason-lspconfig.nvim) wrap this in a custom handler:

```lua
require('mason-lspconfig').setup({
  handlers = {
    -- First handler must be the default handler
    function(server_name)
      require('lspconfig')[server_name].setup({})
    end,

    ['clangd'] = function()
      require('lspconfig').clangd.setup({
        cmd = { 'clangd', '--completion-style=detailed' }
      })
    end,
})
```
